lib/wpcom-error-handler/wpcom-error-handler.php

```diff
@@ -133,16 +147,18 @@ function wpcom_custom_error_handler( $whether_i_may_die, $type, $message, $file,
 	}
 
 	// @ error suppression
-	if ( 0 == error_reporting() ) {
+	// phpcs:ignore WordPress.PHP.DiscouragedPHPFunctions.runtime_configuration_error_reporting -- false positive
+	if ( 0 === error_reporting() ) {
 		$string = '[Suppressed] ' . $string;
 	}
 
 	$backtrace = wpcom_get_error_backtrace( $file, $type );
 
 	if ( ! empty( $_SERVER['HTTP_HOST'] ) && isset( $_SERVER['REQUEST_URI'] ) ) {
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized -- this is OK, the variable will be escaped later
 		$source = $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'];
 	} else {
-		$source = '$ ' . @join( ' ', $GLOBALS['argv'] );
+		$source = '$ ' . join( ' ', empty( $GLOBALS['argv'] ) || ! is_array( $GLOBALS['argv'] ) ? [] : $GLOBALS['argv'] );
```

Comment:
Less beautiful, but does not suppress errors

> Silencing errors is strongly discouraged. Use proper error checking instead.

Reply:
There are two cases when that line could produce a warning or notice:
1. `'argv'` is not in `$GLOBALS` — a notice will be generated (Undefined index)
2. `$GLOBALS['argv']` is not an array — a warning will be generated (Invalid arguments passed)

Even though `$GLOBALS['argv']` is an array when it is defined, it is still a user-controlled variable, and in theory, nothing stops the user from modifying it the way they want.